examples/relay-chat/index.js

```diff
@@ -0,0 +1,340 @@
+      const messageObj = ChatMessage.decode(wakuMessage.payload);
+      ui.renderMessage({
+        ...messageObj,
+        text: bytesToUtf8(messageObj.text),
+      });
+    }
+  );
+
+  ui.setStatus("started", "success");
+
+  const localPeerId = node.libp2p.peerId.toString();
+
+  const remotePeers = await node.libp2p.peerStore.all();
+  const remotePeerIds = new Set(remotePeers.map((peer) => peer.id.toString()));
+
+  ui.setRemotePeer(Array.from(remotePeerIds.keys()));
+
+  node.libp2p.addEventListener("peer:connect", async (event) => {
+    remotePeerIds.add(event.detail.toString());
+    ui.setRemotePeer(Array.from(remotePeerIds.keys()));
+    ui.setRelayMeshInfo(node.relay.gossipSub);
+  });
+
+  node.libp2p.addEventListener("peer:disconnect", (event) => {
+    remotePeerIds.delete(event.detail.toString());
+    ui.setRemotePeer(Array.from(remotePeerIds.keys()));
+    ui.setRelayMeshInfo(node.relay.gossipSub);
+  });
+
+  node.libp2p.addEventListener("peer:identify", (event) => {
+    const peer = event.detail;
+
+    if (!peer.protocols.includes("/webrtc-signaling/0.0.1")) {
+      return;
+    }
+
+    ui.setWebrtcPeer(peer.peerId.toString());
+    ui.setRelayMeshInfo(node.relay.gossipSub);
+  });
+
+  window.node = node;
+
+  return {
+    unsubscribeFromMessages,
+    info: {
+      contentTopic,
+      localPeerId,
+    },
+    sendMessage: async ({ text, nick }) => {
+      if (!text || !nick) {
+        return;
+      }
+
+      const protoMessage = ChatMessage.create({
+        nick,
+        timestamp: Date.now(),
+        text: utf8ToBytes(text),
+      });
+
+      await node.relay.send(Encoder, {
+        payload: ChatMessage.encode(protoMessage).finish(),
+      });
+    },
+    dial: async (multiaddr) => {
+      ui.setStatus("connecting...", "progress");
+      await node.dial(multiaddr);
+      ui.setStatus("connected", "success");
+    },
+    dialWebRTCpeer: async (peerId) => {
+      const peers = await node.libp2p.peerStore.all();
+      const circuitPeer = peers.filter(
+        (p) =>
+          p.protocols.includes("/libp2p/circuit/relay/0.2.0/hop") &&
+          p.protocols.includes("/libp2p/circuit/relay/0.2.0/stop")
+      )[0];
+
+      if (!circuitPeer) {
+        throw Error("No Circuit peer is found");
+      }
+
+      let multiaddr = circuitPeer.addresses.pop().multiaddr;
+      multiaddr = `${multiaddr}/p2p/${circuitPeer.id.toString()}/p2p-circuit/webrtc/p2p/${peerId}`;
+
+      await node.dial(multiaddr);
+      ui.setRelayMeshInfo(node.relay.gossipSub);
+    },
+    ensureWebRTCconnectionInRelayMesh: async () => {
+      const promises = node.libp2p
+        .getConnections()
+        .filter((c) => c.stat.multiplexer === "/webrtc")
+        .map(async (c) => {
+          const outboundStream = node.relay.gossipSub.streamsOutbound.get(
+            c.remotePeer.toString()
+          );
+          const isWebRTCOutbound =
+            outboundStream.rawStream.constructor.name === "WebRTCStream";
```

Comment:
afaik this is the only way to know what type of stream is established in the mesh